Automattic/wp-calypso
# Reader: Fix Post Count Not Appearing

## Proposed Changes

For some reason, `formatNumberCompact` doesn't seem to work. I'd like to propose getting rid of it for two reasons: 

- It's only used in the Reader.
- We have `formatNumber`, which as far as I can see achieves the same thing. In fact, it's already used in more recent code in the Reader - that's why the subscription count works, but not the post count!

## Testing Instructions

**Edit:** turns out you need to use a locale that isn't hardcoded in the `formatNumberCompact` list! For instance, British English.

You can see the difference here: https://wordpress.com/read/feeds/25823

**Before:**
<img width="290" alt="Screenshot 2024-02-25 at 21 09 27" src="https://github.com/Automattic/wp-calypso/assets/43215253/bcf09cd9-707d-48c9-a8da-2f7b6c7de036">

**After:**
<img width="294" alt="Screenshot 2024-02-25 at 21 09 29" src="https://github.com/Automattic/wp-calypso/assets/43215253/f49d5ff2-caa1-4268-9db2-efaeae2a2b02">

cc @eoigal, @davemart-in
